In [1]:
#importing neccessary libraries

import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns

In [9]:
# importing the dataset using pandas

df = pd.read_csv(r"D:/portfolio/healthcare_cleaned.csv")
df.head()

Name  Age  Gender Blood Type Medical Condition  \
0        Aaron Adams   38  Female         O-            Cancer   
1      Aaron Aguirre   36    Male         A-          Diabetes   
2     Aaron Anderson   50  Female         A+            Asthma   
3  Aaron Anderson Md   20  Female         A-      Hypertension   
4       Aaron Archer   47  Female         B-            Cancer   

  Date of Admission              Doctor                 Hospital  \
0        10-03-2021            Norma Li                 Hart LLC   
1        26-02-2023        Katrina Luna           Murray-Shelton   
2        18-12-2020    Kenneth Jennings               Tanner-Cox   
3        28-03-2021         Tammy Perez               Ritter LLC   
4        10-01-2021  Cynthia Villanueva  Montes Case and Mendez,   

  Insurance Provider  Billing Amount  Room Number Admission Type  \
0   UnitedHealthcare     26052.10640          363       Elective   
1   UnitedHealthcare     27087.56055          300      Emergency   
2              Cigna     39804.65862          196         Urgent   
3   UnitedHealthcare     16846.41580          249       Elective   
4           Medicare     10602.07719          108         Urgent   

  Discharge Date   Medication  Test Results  
0     29-03-2021      Lipitor  Inconclusive  
1     13-03-2023      Aspirin  Inconclusive  
2     15-01-2021      Aspirin  Inconclusive  
3     09-04-2021  Paracetamol      Abnormal  
4     17-01-2021  Paracetamol  Inconclusive

# we want to know about the dataset so

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40235 entries, 0 to 40234
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Name                40235 non-null  object 
 1   Age                 40235 non-null  int64  
 2   Gender              40235 non-null  object 
 3   Blood Type          40235 non-null  object 
 4   Medical Condition   40235 non-null  object 
 5   Date of Admission   40235 non-null  object 
 6   Doctor              40235 non-null  object 
 7   Hospital            40235 non-null  object 
 8   Insurance Provider  40235 non-null  object 
 9   Billing Amount      40235 non-null  float64
 10  Room Number         40235 non-null  int64  
 11  Admission Type      40235 non-null  object 
 12  Discharge Date      40235 non-null  object 
 13  Medication          40235 non-null  object 
 14  Test Results        40235 non-null  object 
dtypes: float64(1), int64(2), object(12)
memory usage: 4.6

# Now we want to check for null values

In [11]:
df.isnull().sum()

Name                  0
Age                   0
Gender                0
Blood Type            0
Medical Condition     0
Date of Admission     0
Doctor                0
Hospital              0
Insurance Provider    0
Billing Amount        0
Room Number           0
Admission Type        0
Discharge Date        0
Medication            0
Test Results          0
dtype: int64

#in the above result we can see the 0, it is the total null values its zero so there are no null values.

# Now we want to check if there are any duplicate values

# we can check duplicates only in Name column

In [17]:
df['Name'].duplicated().count()

40235

# Now we know that there no duplicates values

REMINDER
# duplicate values and the Name have been cleaned using power query in Excel

#Now we find the total number of patients

In [18]:
df['Name'].count()

40235

#There are total of 40235 patients.

Now we can find the youngest and oldest patient in the dataset

In [20]:
df['Age'].min()

18

In [ ]:
Total number of people of age 18

In [29]:
df[df['Age'] == 18]['Age'].count()

583

In [ ]:
There are currenlty 583 patients who are 18 years of age

In [21]:
df['Age'].max()

85

In [31]:
df[df['Age']==85]['Age'].count()

564

There are currently 564 patients who are 85 years of age

Now we find the types of medical condition the patients are facing

In [33]:
df['Medical Condition'].unique()

array(['Cancer', 'Diabetes', 'Asthma', 'Hypertension', 'Obesity',
       'Arthritis'], dtype=object)

The patients are suffering from the above conditions

Types of medication different persons use

In [34]:
df['Medication'].unique()

array(['Lipitor', 'Aspirin', 'Paracetamol', 'Ibuprofen', 'Penicillin'],
      dtype=object)

In [ ]:
Types of Blood group

In [36]:
df['Blood Type'].unique()

array(['O-', 'A-', 'A+', 'B-', 'B+', 'AB-', 'O+', 'AB+'], dtype=object)

we want to findout which blood group has been diagnosed with most amount of medications

In [37]:
df.groupby(['Blood Type', 'Medication']).size().reset_index(name = 'count')    # reset_index is used to reset the index

Blood Type   Medication  count
0          A+      Aspirin   1029
1          A+    Ibuprofen   1015
2          A+      Lipitor   1013
3          A+  Paracetamol    995
4          A+   Penicillin    975
5          A-      Aspirin   1028
6          A-    Ibuprofen    982
7          A-      Lipitor   1012
8          A-  Paracetamol   1046
9          A-   Penicillin    955
10        AB+      Aspirin    994
11        AB+    Ibuprofen    993
12        AB+      Lipitor   1059
13        AB+  Paracetamol   1019
14        AB+   Penicillin   1036
15        AB-      Aspirin    992
16        AB-    Ibuprofen    998
17        AB-      Lipitor    980
18        AB-  Paracetamol    984
19        AB-   Penicillin   1047
20         B+      Aspirin   1010
21         B+    Ibuprofen   1000
22         B+      Lipitor   1074
23         B+  Paracetamol    978
24         B+   Penicillin   1003
25         B-      Aspirin   1012
26         B-    Ibuprofen   1028
27         B-      Lipitor    994
28         B-  Paracetamol    972
29         B-   Penicillin   1021
30         O+      Aspirin    981
31         O+    Ibuprofen   1001
32         O+      Lipitor   1014
33         O+  Paracetamol   1007
34         O+   Penicillin    970
35         O-      Aspirin   1009
36         O-    Ibuprofen   1013
37         O-      Lipitor   1010
38         O-  Paracetamol    976
39         O-   Penicillin   1010

Which blood has most medical condition

In [41]:
df.groupby(['Blood Type', 'Medical Condition']).size().reset_index()

Blood Type Medical Condition    0
0          A+         Arthritis  855
1          A+            Asthma  784
2          A+            Cancer  862
3          A+          Diabetes  876
4          A+      Hypertension  825
5          A+           Obesity  825
6          A-         Arthritis  846
7          A-            Asthma  864
8          A-            Cancer  805
9          A-          Diabetes  863
10         A-      Hypertension  848
11         A-           Obesity  797
12        AB+         Arthritis  813
13        AB+            Asthma  869
14        AB+            Cancer  814
15        AB+          Diabetes  864
16        AB+      Hypertension  886
17        AB+           Obesity  855
18        AB-         Arthritis  870
19        AB-            Asthma  793
20        AB-            Cancer  849
21        AB-          Diabetes  835
22        AB-      Hypertension  821
23        AB-           Obesity  833
24         B+         Arthritis  858
25         B+            Asthma  808
26         B+            Cancer  875
27         B+          Diabetes  881
28         B+      Hypertension  800
29         B+           Obesity  843
30         B-         Arthritis  851
31         B-            Asthma  816
32         B-            Cancer  828
33         B-          Diabetes  824
34         B-      Hypertension  850
35         B-           Obesity  858
36         O+         Arthritis  869
37         O+            Asthma  822
38         O+            Cancer  809
39         O+          Diabetes  838
40         O+      Hypertension  823
41         O+           Obesity  812
42         O-         Arthritis  859
43         O-            Asthma  861
44         O-            Cancer  846
45         O-          Diabetes  815
46         O-      Hypertension  825
47         O-           Obesity  812